log/log.go

```diff
@@ -142,9 +143,22 @@ func Debugf(msg string, args ...interface{}) {
 	logger.Debugf(msg, args...)
 }
 
-type nilLogger struct {
+var levelPriorities = map[Level]int{
+	NoLevel:    0,
+	PanicLevel: 1,
+	FatalLevel: 2,
+	ErrorLevel: 3,
+	WarnLevel:  4,
+	InfoLevel:  5,
+	DebugLevel: 6,
 }
 
+func Enabled(l Level) bool {
+	return levelPriorities[logger.Level()] >= levelPriorities[l]
```

Comment:
maybe I am wrong but this might not work:
Let's say we have set the level to info = 5.
if we check with debug the this would pass and it should not.
The numbers should be the other way around.
Debug = 0 -> Panic = 5. Zerolog does this also.
The test is also wrong.

Reply:
Actually it's the opposite, let's say we initialize the logger with a logging level of Info (that is a priority of 5). Then checking if the logger is working/enabled for Debug (priority 6) will lead us to check that 5 >= 6 which is false, so for Debug logging level our logger would not log. The same check will return true for logging levels of eg Info, Warn, does that make sense?

PS: About the priority numbers, it's in an order meaning that the lower number has a higher priority, like for example process priority in UNIX OS'es.

Reply:
yes, I think we are saying the same thing. The difference is that zerolog has the order reversed (https://github.com/rs/zerolog/blob/master/log.go) and the check obviously also. 
The same goes for Uber zap logger (https://github.com/uber-go/zap/blob/master/zapcore/level.go) and for Google's glog (https://github.com/golang/glog/blob/master/glog.go).
Logrus, on the other hand, follows the order you are proposing (https://github.com/sirupsen/logrus/blob/master/logrus.go). 
I am more accustomed to zerolog's, zap's and glog's approach saying `if my level is higher than the default one, meaning more important because of a bigger number, the allow me to log.`.
If it is not a problem to change the order, but I will not insist on it.

Reply:
No problem, we have just to reverse the `>=` to `<=` (and also to change some numbers :P)